pkg/datastore/firestore/iterator.go

```diff
@@ -33,10 +37,32 @@ func (it *Iterator) Next(dst interface{}) error {
 		}
 		return err
 	}
+
+	// Update last iterated item as last read doc.
+	it.last = doc
+
 	return doc.DataTo(dst)
 }
 
+// Cursor builds a string (in map[string]interface{} format).
+// The cursor contains only values attached with the fields used
+// as ordering fields.
 func (it *Iterator) Cursor() (string, error) {
-	// Note: Cursor function is not needed in Cloud Firestore.
-	return "", datastore.ErrUnimplemented
+	if it.last == nil {
+		return "", datastore.ErrInvalidCursor
+	}
+
+	lastObjData := it.last.Data()
+
+	cursor := make(map[string]interface{})
+	for _, o := range it.orders {
+		val, ok := lastObjData[o.Field]
+		if !ok {
+			return "", datastore.ErrInvalidCursor
+		}
+		cursor[o.Field] = val
+	}
+
+	b, _ := json.Marshal(cursor)
+	return string(b), nil
```

Comment:
Could you add a test to ensure the whole flow around this?
For example, making a dummy application model and then convert it to firestore.DocSnapshot to set to the Iterator.
Then try to retrieve the Cursor string and use the `makeCursorValues` to ensure that the returned data is correct.

Reply:
Addressed on https://github.com/pipe-cd/pipe/pull/1864/commits/6178f46887eef3810fb9de1101772d3af96a9417 🙏